lib/ClangImporter/ImporterImpl.h

```diff
@@ -1891,43 +1891,20 @@ static inline Type applyToFunctionType(
   return type;
 }
 
-inline Optional<const clang::EnumDecl *> findAnonymousEnumForTypedef(
-    const ASTContext &ctx,
-    const clang::TypedefType *typedefType) {
-  auto *typedefDecl = typedefType->getDecl();
-  auto *lookupTable = ctx.getClangModuleLoader()->findLookupTable(typedefDecl->getOwningModule());
-
-  auto foundDecls = lookupTable->lookup(
-      SerializedSwiftName(typedefDecl->getName()), EffectiveClangContext());
-
-  auto found = llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-    return decl.is<clang::NamedDecl *>() &&
-        isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>());
-  });
-
-  if (found != foundDecls.end())
-    return cast<clang::EnumDecl>(found->get<clang::NamedDecl *>());
-
-  // If a swift_private attribute has been attached to the enum, its name will
-  // be prefixed with two underscores
-  llvm::SmallString<32> swiftPrivateName;
-  swiftPrivateName += "__";
-  swiftPrivateName += typedefDecl->getName();
-  foundDecls = lookupTable->lookup(
-      SerializedSwiftName(ctx.getIdentifier(swiftPrivateName)),
-      EffectiveClangContext());
-
-  auto swiftPrivateFound =
-      llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-        return decl.is<clang::NamedDecl *>() &&
-               isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>()) &&
-               decl.get<clang::NamedDecl *>()
-                   ->hasAttr<clang::SwiftPrivateAttr>();
-      });
-
-  if (swiftPrivateFound != foundDecls.end())
-    return cast<clang::EnumDecl>(swiftPrivateFound->get<clang::NamedDecl *>());
-
+inline Optional<const clang::EnumDecl *>
+findAnonymousEnumForTypedef(const ASTContext &ctx,
+                            const clang::TypedefType *typedefType) {
+
+  assert(typedefType && "Must provide non-null typedef");
+  const clang::TypedefNameDecl *typedefDecl = typedefType->getDecl();
+  const clang::DeclContext *declContext = typedefDecl->getDeclContext();
+  for (const clang::Decl *siblingDecl : declContext->decls()) {
+    if (const clang::EnumDecl *enumDecl =
+            dyn_cast<clang::EnumDecl>(siblingDecl)) {
+      if (enumDecl->getDeclName().isEmpty() && enumDecl->getIntegerType().getTypePtrOrNull() == typedefType)
+        return enumDecl;
```

Comment:
This could find any enum, if you want to use this approach you should check the source location is the same as the macro.

Reply:
I think it would only find enums backed by the typedef in question, but I will try to check the source location.